client/cli/src/commands/export_sync_state_cmd.rs

```diff
@@ -0,0 +1,96 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::error;
+use crate::params::{SharedParams, NetworkParams};
+use crate::CliConfiguration;
+use crate::Role;
+use log::info;
+use sc_service::ChainSpec;
+use sc_service::chain_ops::{MaybeChtRootStorageProvider, build_light_sync_state};
+use structopt::StructOpt;
+use std::io::Write;
+use std::sync::Arc;
+use std::task::Poll;
+use sp_runtime::traits::Block as BlockT;
+
+/// Export the chain spec containing a state to sync the light client.
+#[derive(Debug, StructOpt)]
+pub struct ExportSyncStateCmd {
```

Comment:
Why is that a new CLI command?

Why isn't that just part of `build-spec` as cli argument? 

I don't really see the reason for a new command.